docs/rfcs/1013-parsec-account.md

```diff
@@ -0,0 +1,507 @@
+body\_sha256 = sha256(body) \\
+content = \text{"PARSEC-HMAC-BLAKE2B"} | email | timestamp | body\_sha256 \\
+code = hmac_{black2b}(shared\_secret, content) \\
+signature = base64(code)
+\end{gather}
+```
+
+### The data model
+
+#### Account
+
+[`Account`]: #account
+
+```yaml
+{
+  "label": "Account",
+  "type": "account",
+  "other_fields": {
+    "email": "EmailStr",
+    "label": "EmailLabel"
+  }
+}
+```
+
+The `label` is not used by Parsec Account server but will be used by the client to pre-fill the device label information (and the email field).
+
+#### Vault
+
+[`Vault`]: #vault
+
+```yaml
+{
+  "label": "Vault",
+  "type": "vault",
+  "other_fields": {
+    "created_on": "Timestamp",
+    "created_by": "Creator"
+  }
+}
+```
+
+> `Creator` is a string-like value containing IP address and User-Agent to
+> identify the creator of the vault.
+- There may be multiple vaults for an account.
+- For any given account, only a single vault is active.
+- The older vaults are kept for recovery purposes.
+
+#### Auth Method
+
+[`AuthMethod`]: #auth-method
+
+```yaml
+{
+  "label": "AuthMethod",
+  "type": "auth_method",
+  "other_fields": {
+    "algorithm": "KeyAlgorithm",
+    "hmac_key": "HMACKey",
+    "account_id": "AccountID",
+    "vault_id": "VaultID",
+    "vault_encrypted_key": "EncryptedSymKey",
+    "created_on": "Timestamp",
+    "created_by": "Creator"
+  }
+}
+```
+- The algorithm field contains the full configuration needed to do the derivations operations. Typically, for the password it should be `{"name": "Argon2iBlake2", "argon2i_opslimit": 65536, "argon2i_memlimit_kb": 3, "argon2i_parallelism": 1}`
+- There can be only a single password authentication method per vault.
+- Other authentication methods (such as FIDO2) may be added in the future.
+
+#### Protected Device
+
+[`ProtectedDevice`]: #protected-device
+
+```yaml
+{
+  "label": "ProtectedDevice",
+  "type": "protected_device",
+  "other_fields": {
+    "organization_id": "OrganizationID",
+    "device_id": "DeviceID",
+    "vault_id": "ID",
+    "encrypted_device": "EncryptedDevice"
+  }
+}
+```
+- Device are protected within a vault
+- Any given device can be protected in multiple vaults (this is typically the case when the devices have been recovered from an older vault).
+
+### API
+
+#### Account creation
+
+To create an account, the client starts by requesting an email validation token.
+
+```yaml
+{
+  "cmd": "account_send_email_validation_token",
+  "req": {
+    "email": "EmailStr"
+  },
+  "reps": [
+    {
+      "status": "ok"
+    },
+    {
+      "status": "invalid_email"
+    }
+  ]
+}
+```
+
+On `ok`, the server would have sent a mail with a unique token used for next the request used to register the authentication method.
+
+> The token will be valid for a duration that will be defined at the implementation (likely will be configurable)
+
+If the email is already registered in the service, a `ok` response will still be sent (without sending an email)
```

Comment:
It's a remark for later, but should we send an email to warn the user that another account creation attempt has occurred ?

Reply:
When writing the RFC I though of the same thing but I do not see how actionable the email would be to the user.